src/config/kubeconfig.js

```diff
@@ -1,9 +1,73 @@
 'use strict';
 
-const process = require('process');
 const pth = require('path');
 const fs = require('fs');
-const k8s = require('@kubernetes/client-node');
+// TODO: Uncomment after PR https://github.com/kubernetes-client/javascript/pull/748 lands
+// const k8s = require('@kubernetes/client-node');
+
+// TODO: Delete next two functions
+// after PR https://github.com/kubernetes-client/javascript/pull/748 lands
+
+function dropDuplicatesAndNils(a) {
+  return a.reduce((acceptedValues, currentValue) => {
+    // Good-enough algorithm for reducing a small (3 items at this point) array into an ordered list
+    // of unique non-empty strings.
+    if (currentValue && !acceptedValues.includes(currentValue)) {
+      return acceptedValues.concat(currentValue);
+    } else {
+      return acceptedValues;
+    }
+  }, []);
+}
+
+// Only public for testing.
+function findHomeDir() {
+  if (process.platform !== 'win32') {
+    if (process.env.HOME) {
+      try {
+        fs.accessSync(process.env.HOME);
+
+        return process.env.HOME;
+        // tslint:disable-next-line:no-empty
+      } catch (ignore) { }
+    }
+
+    return null;
+  }
+  const homeDrivePath = process.env.HOMEDRIVE && process.env.HOMEPATH ? pth.join(process.env.HOMEDRIVE, process.env.HOMEPATH) : null;
+  const dirList1 = dropDuplicatesAndNils([process.env.HOME, homeDrivePath, process.env.USERPROFILE]);
+  const dirList2 = dropDuplicatesAndNils([process.env.HOME, process.env.USERPROFILE, homeDrivePath]);
```

Comment:
What's the different between the two lists? Please have better names (why are they needed?).

Reply:
Changed the names to reflect the way the preferred order changes with context -- the comments already explain those.